web/src/components/storage/BootSelectionDialog.jsx

```diff
@@ -0,0 +1,168 @@
+const BOOT_DISABLED_ID = "boot-disabled";
+const OPTIONS_NAME = "boot-mode";
+
+/**
+ * Internal component for building the options
+ * @component
+ *
+ * @param {React.PropsWithChildren<React.ComponentProps<"input">>} props
+ */
+const RadioOption = ({ id, onChange, defaultChecked, children }) => {
+  return (
+    <>
+      <input id={id} name={OPTIONS_NAME} type="radio" defaultChecked={defaultChecked} onChange={onChange} />
+      <label htmlFor={id}>{children}</label>
+    </>
+  );
+};
+
+/**
+ * Renders a dialog that allows the user to select the boot configuration.
+ * @component
+ *
+ * @typedef {object} Boot
+ * @property {boolean} configureBoot
+ * @property {StorageDevice|undefined} bootDevice
+ *
+ * @param {object} props
+ * @param {boolean} props.configureBoot - Whether the boot is configurable
+ * @param {StorageDevice|undefined} props.bootDevice - Currently selected booting device.
+ * @param {StorageDevice[]} props.devices - Devices that user can select to boot from.
+ * @param {boolean} [props.isOpen=false] - Whether the dialog is visible or not.
+ * @param {function} [props.onCancel=noop] - Callback to execute when user closes the dialog.
+ * @param {(boot: Boot) => void} props.onAccept
+ */
+export default function BootSelectionDialog({
+  configureBoot: defaultConfigureBoot,
+  bootDevice: defaultBootDevice,
+  devices,
+  isOpen,
+  onCancel = noop,
+  onAccept = noop,
+  ...props
+}) {
+  const [configureBoot, setConfigureBoot] = useState(defaultConfigureBoot);
+  const [bootDevice, setBootDevice] = useState(defaultBootDevice || devices[0]);
+  const [isBootAuto, setIsBootAuto] = useState(defaultConfigureBoot && defaultBootDevice === undefined);
+
+  const isBootManual = configureBoot && !isBootAuto;
+
+  const selectBootAuto = () => {
+    setConfigureBoot(true);
+    setIsBootAuto(true);
+  };
+
+  const selectBootManual = () => {
+    setConfigureBoot(true);
+    setIsBootAuto(false);
+  };
+
+  const selectBootDisabled = () => {
+    setConfigureBoot(false);
+    setIsBootAuto(false);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const device = isBootAuto ? undefined : bootDevice;
+    onAccept({ configureBoot, bootDevice: device });
+  };
+
+  const isAcceptDisabled = () => {
+    return isBootManual && bootDevice === undefined;
+  };
+
+  return (
+    <Popup
+      title={_("Configuration for boot partitions")}
+      isOpen={isOpen}
+      className="large"
+      {...props}
+    >
+      <Form id="boot-form" onSubmit={onSubmit}>
+        <fieldset className="stack">
+          <legend className="split">
+            <RadioOption id={BOOT_AUTO_ID} defaultChecked={isBootAuto} onChange={() => selectBootAuto()}>
+              {_("Automatic")}
+            </RadioOption>
+          </legend>
+          <div>
+            {_("Additional partitions to boot the system will be configured at /dev/vdc, \
+            the device selected for installing the system.")}
```

Comment:
Actually, the idea here is to go for a `%s` and then replace it by the selected _installationDevice_

```suggestion
            {_("Additional partitions to boot the system will be configured in \
            the device selected for installing the system, currently %s")}
```

But I'm not sure how to make it work. I mean, when _target_ is disk is pretty clear, just to put the _device.name_ or _deviceLabel_ there. But what is the right value when the _target_ is a new LVM thing (see https://github.com/openSUSE/agama/pull/1106/files#diff-bd6c15ca365f9adcb6bb3d76300f7729cfbd21274b895e1504c5956443e8a99cR53-R61)

Let's talk about it on Monday.